### ISSUE TITLE
v2 feature: equivalents

### ISSUE THREAD

**Issue by augustjohnson** [open5e/open5e-api#461]
For a given model, include an "equivalents" many-to-many relationship where a given object can give a list of bi-directionally related to "equivalent" objects from different documents.

Size-Large from the SRD is technically distinct from Size-Large from the A5e SRD, but they are equivalents.

**Comment by eepMoody** [comment]
Proposal: alternate rules for properties/concept (acid, large, fey, invisible) are encoded as sub-objects underneath the concept.

There should be a filter for "has a definition in these rulesets" (eg. srd-2024, srd-2012, a5e)

```
Dazed
  2012
    Desc
    Doc
  A5e
     Desc
     Doc
  2024
     Desc
     Doc
```

**Comment by eepMoody** [comment]
Should there be a query parameter for a "game system"? Have a hierarchy (ordered list) of SRDs and fallback to the higher-tier SRDs if a lower one is requested but unavailable.

The actual description is an object that's underneath, and we surface that based on a query parameter.

We will need a new data object to handle this, but from what's presented this will not be substantially different.

**Comment by augustjohnson** [comment]
I think where I landed on this one was that a single "abberations" would continue to exist, but be a distinct concept in the API, and it would have that name, but it would pull the description based on a querystring parameter (in a preferred order of SRDs). I think we also concluded that this was not a requirement for the launch of v2.

**Comment by eepMoody** [comment]
Agreed. @augustjohnson Do you mind if I update the description to reflect that decision?

**Comment by augustjohnson** [comment]
Branch created. Brainstorming on the detailed solution below.

1. We need a word for the concepts that are fairly core to 5e.
2. We need a priority for which desc to load when no desc is specified.
3. We need the list of concepts that this "equivalent" applies to.

Word options: 
- "Core" - Core Creature Types - sounds ok, but wizards uses "core rules" as the description for the new SRD so that's confusing.
- "Standard" - Standard Damage Type - Not bad, but I don't love it. It implies non-standard, which is a bit weird.
- "5e" - 5eAbility - Could be ok, but I'm not sure database stuff allows starting with a number.
- "Common" - Common Alignment - Pretty good, but does get confusing with Language. 
- "Shared" - Shared Size - Feels awkward, but is fairly descriptive for how it's expected to be used, which is a nice feature.

Priority:
I'd say we just hand-prioritize basically.
SRD 2014 if unspecified or the specified fails.
At v2 launch, move to SRD2024.

Models List:

- Creature Types
- Damage Types
- Item Category
- Item Rarity
- Size
- Skill
- Ability
- Environment(?)
- Alignment
- Language(?)
- SpellSchool
- Weapon Property

**Comment by augustjohnson** [comment]
After a bit more design and thought, I think there are a set of models which are CORE models (roughly the list above), and those should basically be enums, functionally. Then each of those core model objects, when being loaded with an argument of a document, can present a field (desc usually) based on that argument.

These are also excellent candidate models for object icons.

**Comment by augustjohnson** [comment]
Ok, here's the design.
Create a new DOCUMENT that describes these core concepts, such as damagetype:acid.
Migrate all existing items into that document.
From each of the models above, remove the "HasDescription" statement.
Create new models associated with "DamageTypeDescription and move the descriptions into those.
Adjust the DamageType model to have a description lookup that chooses a default, but can be overridden.

**Comment by augustjohnson** [comment]
- [x]     Creature Types
- [x]     Language
- [x]     Damage Types
- [x]     Item Category
- [x]     Item Rarity
- [x]     Size
- [x]     Skill
- [x]     Ability
- [x]     Environment(?)
- [x]     Alignment
- [x]     SpellSchool
- [ ]     Weapon Property

**Comment by augustjohnson** [comment]
I am choosing not to move weapon properties until further discussion occurs.
Follow up task: For some models above, the "description" field should be extended to produce a distinct description based on some sort of document selected.

**Comment by augustjohnson** [comment]
Damage Types - SRD-2024 does not have a description (only examples) for each damage type. Consider an override.
Languages - No override for description there, it's pretty universal.
Creature Types - These do have a distinct description, and should have an override.
Item Category - Not high priority to have this described distinctly.
Item Rarity - does not have a description in 2024.
Size - No description.
Skill - 2024 has a field titled "Example Uses" for this, which should probably just be part of the description field. Yes, we should allow for overrides.
Ability - Definitely should have an override, but is functionally pretty similar. Field is titled "Make a Check to..."
Environment - Never had a description in any SRD.
Alignment - Has a different description per document. Should have an override.
SpellSchool - The column is called "Typical Effects" now, and is pretty similar but not quite the same as the previous description.

**Comment by augustjohnson** [comment]
- [x] DamageTypeDescription exists and DamageType has response parity.
- [x] CreatureTypeDescription exists, response parity
- [x] SkillDescription exists, response parity
- [x] AbilityDescription exists, response parity
- [x] AlignmentDescription exists, response parity.
- [x] Expose "description document" field to the api response for all above.
- [ ] Add in data for 2024 srd for all of the above.

**Comment by augustjohnson** [comment]
Added #790 for a follow up task.
